clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -478,7 +482,7 @@ public void abortIncompleteBatches() {
      */
     private void abortBatches() {
         for (RecordBatch batch : incomplete.all()) {
-            Deque<RecordBatch> dq = dequeFor(batch.topicPartition);
+            Deque<RecordBatch> dq = getDeque(batch.topicPartition);
```

Comment:
I debated adding the null check. I finally decided against it because the method is private and the class invariants seem to preclude a null in this case (the only time we remove deques is in shutdown). I was a little torn, so let me know if anyone has a preference.